src/zenml/zen_stores/schemas/run_metadata_schemas.py

```diff
@@ -0,0 +1,155 @@
+    """SQL Model for run metadata."""
+
+    __tablename__ = "run_metadata"
+
+    pipeline_run_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=PipelineRunSchema.__tablename__,
+        source_column="pipeline_run_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=True,
+    )
+    pipeline_run: "PipelineRunSchema" = Relationship(
+        back_populates="run_metadata"
+    )
+
+    step_run_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=StepRunSchema.__tablename__,
+        source_column="step_run_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=True,
+    )
+    step_run: "StepRunSchema" = Relationship(back_populates="run_metadata")
+
+    artifact_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=ArtifactSchema.__tablename__,
+        source_column="artifact_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=True,
+    )
+    artifact: "ArtifactSchema" = Relationship(back_populates="run_metadata")
+
+    stack_component_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=StackComponentSchema.__tablename__,
+        source_column="stack_component_id",
+        target_column="id",
+        ondelete="SET NULL",
+        nullable=True,
+    )
+    stack_component: "StackComponentSchema" = Relationship(
+        back_populates="run_metadata"
+    )
+
+    user_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=UserSchema.__tablename__,
+        source_column="user_id",
+        target_column="id",
+        ondelete="SET NULL",
+        nullable=True,
+    )
+    user: "UserSchema" = Relationship(back_populates="run_metadata")
+
+    project_id: UUID = build_foreign_key_field(
+        source=__tablename__,
+        target=ProjectSchema.__tablename__,
+        source_column="project_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=False,
+    )
+    project: "ProjectSchema" = Relationship(back_populates="run_metadata")
+
+    key: str
+    value: str = Field(sa_column=Column(TEXT, nullable=False))
+    type: MetadataTypeEnum
+
+    def to_model(self) -> "RunMetadataResponseModel":
+        """Convert a `RunMetadataSchema` to a `RunMetadataResponseModel`.
+
+        Returns:
+            The created `RunMetadataResponseModel`.
+        """
+        return RunMetadataResponseModel(
+            id=self.id,
+            pipeline_run_id=self.pipeline_run_id,
+            step_run_id=self.step_run_id,
+            artifact_id=self.artifact_id,
+            stack_component_id=self.stack_component_id,
+            key=self.key,
+            value=json.loads(self.value),
+            type=self.type,
+            project=self.project.to_model(),
+            user=self.user.to_model(),
```

Comment:
Should take into account that users could be deleted while the metadata stays

```suggestion
            user=self.user.to_model() if self.user else None,
```

Reply:
Good catch!